packages/components/Menu/src/MenuItem/useMenuItem.ts

```diff
@@ -52,6 +52,23 @@ export const useMenuItem = (props: MenuItemProps): MenuItemInfo => {
   );
 
   const pressable = usePressableState({ ...rest, onPress: onInvoke });
+  const { onBlur, onHoverOut, ...restPressableProps } = pressable.props;
+  const onHoverOutModified = React.useCallback(
+    (e) => {
+      // Focus is removed on macOS while win32 keeps it
+      if (Platform.OS === 'macos') {
+        onBlur(e);
+      }
+      onHoverOut && onHoverOut(e);
+    },
+    [onBlur, onHoverOut],
+  );
+
+  const pressablePropsModified = {
+    onBlur,
+    onHoverOut: onHoverOutModified,
+    ...restPressableProps,
+  };
```

Comment:
Could we do this before `usePressableState` do that we don't have to modify pressable.props?
I did that in ContextualMenu because we had to do a lot of shennanigans of wanting both `onMouseEnter` and `onHoverIn`, but it doesn't seem like that applies here.

Reply:
Is there a way where I can override `onBlur` and `onHoverOut` without modifying pressable.props?

Reply:
Passing on our / onHoverOut into usePressableState doesn't work?

Reply:
Passing on our / onHoverOut into usePressableState doesn't work?

Reply:
I don't think so since onBlur is called inside the onHoverOut override?